src/ucp/core/ucp_request.h

```diff
@@ -319,19 +319,17 @@ struct ucp_request {
  */
 struct ucp_recv_desc {
     union {
-        ucs_list_link_t     tag_list[2];    /* Hash list TAG-element */
-        ucs_queue_elem_t    stream_queue;   /* Queue STREAM-element */
-        ucs_queue_elem_t    tag_frag_queue; /* Tag fragments queue */
+        ucs_list_link_t     tag_list[2];     /* Hash list TAG-element */
+        ucs_queue_elem_t    stream_queue;    /* Queue STREAM-element */
+        ucs_queue_elem_t    tag_frag_queue;  /* Tag fragments queue */
     };
-    uint32_t                length;         /* Received length */
-    uint32_t                payload_offset; /* Offset from end of the descriptor
-                                             * to AM data */
-    uint16_t                flags;          /* Flags */
-    int16_t                 priv_length;    /* Number of bytes consumed from
-                                               headroom private space, except the
-                                               space needed for ucp_recv_desc itself.
-                                               It is used for releasing descriptor
-                                               back to UCT only */
+    uint32_t                length;          /* Received length */
+    uint32_t                payload_offset;  /* Offset from end of the descriptor
+                                              * to AM data */
+    uint16_t                flags;           /* Flags */
+    int32_t                 uct_desc_offset; /* Offset which needs to be
```

Comment:
uint16_t

Reply:
made it signed specifically, otherwise during release when do
`-rdesc->uct_desc_offset`

Reply:
1. still need to make it 16-bit to not increase rdesc
2. are you sure that need to cast? if yes, it's better IMHO, since the value in this struct is always >= it should be unsigned

Reply:
1) It does not increase, we had some space (I checked). Can do i16 though
2)  Yes, spent some time investigating weird crashes yesterday. We have an assert checking it is positive, so I'd avoid extra casting, which looks confusing.

Reply:
ok, so i16

Reply:
ok